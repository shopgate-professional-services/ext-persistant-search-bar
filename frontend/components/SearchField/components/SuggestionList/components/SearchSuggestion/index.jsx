import React from 'react';
import PropTypes from 'prop-types';
import List from '../List';
import styles from '../../style';

/**
 * Search suggestion list entry component.
 * @param {Object} props The component props.
 * @param {string} props.suggestion The search suggestion text.
 * @param {Function} props.onClick The action that is triggered on click.
 * @returns {JSX}
 */
const SearchSuggestion = ({ suggestion, onClick }) => (
  <List.Item
    className={styles.listItem}
    title={suggestion}
    onClick={onClick}
  />
);

SearchSuggestion.propTypes = {
  onClick: PropTypes.func.isRequired,
  suggestion: PropTypes.string.isRequired,
};

export default SearchSuggestion;
