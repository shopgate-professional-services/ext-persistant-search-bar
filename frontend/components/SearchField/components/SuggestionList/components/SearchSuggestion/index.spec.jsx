import React from 'react';
import { mount } from 'enzyme';
import SearchSuggestion from './index';

describe('<SearchSuggestion', () => {
  it('should render', () => {
    const component = mount(<SearchSuggestion suggestion="foo" onClick={() => {}} />);
    expect(component).toMatchSnapshot();
  });
});
