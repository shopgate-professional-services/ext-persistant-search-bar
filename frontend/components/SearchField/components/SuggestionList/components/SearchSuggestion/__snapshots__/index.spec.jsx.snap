// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchSuggestion should render 1`] = `
<SearchSuggestion
  onClick={[Function]}
  suggestion="foo"
>
  <Item
    className={null}
    image={null}
    isDisabled={false}
    isSelected={false}
    link={null}
    onClick={[Function]}
    rightComponent={null}
    testId={null}
    title="foo"
  >
    <div
      aria-hidden={true}
      data-test-id={null}
      onClick={[Function]}
    >
      <ForwardRef(WithForwardedRef(Glow))
        className={null}
      >
        <Glow
          className={null}
          color="#f7f7f7"
          forwardedRef={null}
          styles={
            Object {
              "container": null,
              "glow": null,
              "hover": null,
            }
          }
        >
          <div
            className="css-13azwyo"
            onClick={[Function]}
            style={null}
          >
            <div
              className="css-1bmd5dz"
              style={Object {}}
            />
            <Grid
              className="css-1qcr7hz"
              component="div"
              wrap={false}
            >
              <div
                className="css-1qcr7hz css-h7r55a"
              >
                <div
                  className="css-jwxytc"
                />
                <GridItem
                  className="css-10fxwvh"
                  component="div"
                  grow={1}
                  shrink={1}
                >
                  <div
                    className="css-10fxwvh css-1gs0ko2"
                  >
                    foo
                  </div>
                </GridItem>
              </div>
            </Grid>
          </div>
        </Glow>
      </ForwardRef(WithForwardedRef(Glow))>
    </div>
  </Item>
</SearchSuggestion>
`;
